### PR TITLE
[tests-only][full-ci] added test to send share invitation to multiple groups at once

### PR DESCRIPTION
**Pull request by PrajwolAmatya**
## Description
Added test to send share invitation to multiple groups at once.
Expected: sharing should fail with status code `400`

## Related Issue

- https://github.com/owncloud/ocis/issues/7947

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Review by saw-jan — APPROVED** [review]
LGTM

File: tests/acceptance/features/apiSharingNg/shareInvitations.feature [hunk 1]
```diff
@@ -774,3 +774,70 @@ Feature: Send a sharing invitations
...
+      | Carol    | grp2      |
+      | Bob      | grp2      |
+    And user "Alice" has uploaded file with content "to share" to "/textfile1.txt"
+    And user "Alice" has created folder "FolderToShare"
+    When user "Alice" sends the following share invitation using the Graph API:
+      | resourceType    | <resource-type>    |
+      | resource        | <path>             |
+      | space           | Personal           |
+      | sharee          | grp1, grp2         |
+      | shareType       | group, group       |
```
Comment (grgprarup) [inline tests/acceptance/features/apiSharingNg/shareInvitations.feature:798]:
```suggestion
      | shareType       | group       |
```

File: tests/acceptance/features/bootstrap/SharingNgContext.php [hunk 2]
```diff
@@ -142,9 +142,15 @@ public function userSendsTheFollowingShareInvitationUsingTheGraphApi(string $use
 			? $this->spacesContext->getResourceId($user, $rows['space'], $rows['resource'])
 			: $this->spacesContext->getFileId($user, $rows['space'], $rows['resource']);
 
-		$shareeId = ($rows['shareType'] === 'user')
-			? $this->featureContext->getAttributeOfCreatedUser($rows['sharee'], 'id')
-			: $this->featureContext->getAttributeOfCreatedGroup($rows['sharee'], 'id');
+		$sharee = array_map('trim', explode(',', $rows['sharee']));
```
Comment (grgprarup) [inline tests/acceptance/features/bootstrap/SharingNgContext.php:145]:
```suggestion
		$sharees = array_map('trim', explode(',', $rows['sharee']));
```

File: tests/acceptance/features/bootstrap/SharingNgContext.php [hunk 3]
```diff
@@ -142,9 +142,15 @@ public function userSendsTheFollowingShareInvitationUsingTheGraphApi(string $use
...
+		$shareType = array_map('trim', explode(',', $rows['shareType']));
+
+		$shareeId = [];
```
Comment (grgprarup) [inline tests/acceptance/features/bootstrap/SharingNgContext.php:148]:
```suggestion
		$shareeIds = [];
```

File: tests/acceptance/features/bootstrap/SharingNgContext.php [hunk 4]
```diff
@@ -142,9 +142,15 @@ public function userSendsTheFollowingShareInvitationUsingTheGraphApi(string $use
...
+		for ($i = 0; $i < \count($sharee); $i++) {
+			$shareeId[] = ($shareType[$i] === 'user')
+				? $this->featureContext->getAttributeOfCreatedUser($sharee[$i], 'id')
+				: $this->featureContext->getAttributeOfCreatedGroup($sharee[$i], 'id');
+		}
```
Comment (grgprarup) [inline tests/acceptance/features/bootstrap/SharingNgContext.php:153]:
```suggestion
		foreach ($sharees as $sharee) {
			$shareeIds[] = ($shareType === 'user')
				? $this->featureContext->getAttributeOfCreatedUser($sharee, 'id')
				: $this->featureContext->getAttributeOfCreatedGroup($sharee, 'id');
		}
```

File: tests/acceptance/features/apiSharingNg/shareInvitations.feature (hunk 1 above)
Comment (PrajwolAmatya) [reply tests/acceptance/features/apiSharingNg/shareInvitations.feature:798]: It is added just to differentiate between share type, for instance, if the resource is shared to group and user at once.

File: tests/TestHelpers/GraphHelper.php [hunk 5]
```diff
@@ -1598,19 +1598,21 @@ public static function sendSharingInvitation(
...
-		$recipients['objectId'] = $shareeId;
-		$recipients['@libre.graph.recipient.type'] = $shareType;
-
-		$body['recipients'] = [$recipients];
+		for ($i = 0; $i < \count($shareeIds); $i++) {
+			$body['recipients'][] = [
+				"@libre.graph.recipient.type" => $shareTypes[$i],
+				"objectId" => $shareeIds[$i]
+			];
+		}
```
Comment (grgprarup) [inline tests/TestHelpers/GraphHelper.php:1615]:
```suggestion
		foreach ($shareeIds as $index => $shareeId) {
		        $shareType = $shareTypes[$index];
			$body['recipients'][] = [
				"@libre.graph.recipient.type" => $shareType,
				"objectId" => $shareeId
			];
		}
```
Can we do in this way?

**Pull request merged**